### PR TITLE
Workflow

### PR DESCRIPTION
**Pull request by Lodoen**

### CODE REVIEW

**Comment by Lodoen** [comment]
Requests review.

**Review by Fermain — CHANGES_REQUESTED** [review]
This is very good work, although there is one file missing. A static pages deploy with `npm build` is the unmet requirement.

**Review by Fermain — APPROVED** [review]

File: src/js/api/auth/login.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+global.localStorage = new LocalStorageMock();
+
+describe("login", () => {
+  it("stores a token when provided with valid credentials", async () => {
+    localStorage.clear();
+    await login("workflowca@stud.noroff.no", "");
+    const token = localStorage.getItem("token");
+    expect(token).not.toBeNull();
```
Comment (Fermain) [inline src/js/api/auth/login.test.js:41]: You could `expect(fetch).toHaveBeenCalledTimes(1)` to harden this

File: src/js/api/posts/create.test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  ok: true,
+  json: jest.fn().mockResolvedValue(mockObject),
+});
+
+global.fetch = mockFetchSuccess;
+
+describe("create", () => {
+  it("returns a valid item with a valid input", async () => {
+    const post = await createPost("title", "body", "", ["1", "2"]);
+    expect(post).toMatchObject(mockObject);
```
Comment (Fermain) [inline src/js/api/posts/create.test.js:27]: Same here `expect(fetch).toHaveBeenCalledTimes(1)`

**Review by Fermain — APPROVED** [review]
This is a good submission without any major issues or concerns regarding the assessment criteria. Your cypress and jest tests show some lateral thinking and an understanding of the principles of this difficult subject. Enjoy your weekend!

File: src/js/api/auth/login.test.js (hunk 1 above)
Comment (Lodoen) [reply src/js/api/auth/login.test.js:41]: This has now been implemented.

File: src/js/api/posts/create.test.js (hunk 2 above)
Comment (Lodoen) [reply src/js/api/posts/create.test.js:27]: This has now been implemented.

**Review by Fermain — APPROVED** [review]
A good submission that shows confidence with workflow tooling.

**Pull request merged**